exec/cnex/nstring.c

```diff
@@ -221,6 +233,35 @@ BOOL string_startsWith(TString *self, TString *s)
     return TRUE;
 }
 
+BOOL string_endsWithNoCase(TString *self, TString *s)
+{
+    size_t i, si;
+
+    for (si = 0, i = self->length - s->length; i < self->length; si++, i++) {
```

Comment:
This will fail with undefined behavior if `s->length` is greater than `self->length`.

Reply:
Added pre-check to both functions.